src/features/query-bar/components/filter/Condition.js

```diff
@@ -68,36 +67,39 @@ const Condition = ({
     };
 
     const updateSelectedField = (option: SelectOptionType, fieldType?: string) => {
-        const conditionIndex = index;
-        const value = option.value;
-        const valueType = option.type;
-        const fieldId = option.fieldId || condition.fieldId;
+        const { displayText, fieldId, type, value } = option;
 
-        const displayText = option.displayText;
         let displayTextType = '';
         let keyType = '';
 
-        if (fieldType === COLUMN) {
-            displayTextType = COLUMN_DISPLAY_TEXT;
-            keyType = COLUMN_KEY;
-        } else if (fieldType === OPERATOR) {
-            displayTextType = OPERATOR_DISPLAY_TEXT;
-            keyType = OPERATOR_KEY;
-        } else if (fieldType === VALUE) {
-            displayTextType = VALUE_DISPLAY_TEXT;
-            keyType = VALUE_KEY;
+        switch (fieldType) {
+            case COLUMN:
+                displayTextType = COLUMN_DISPLAY_TEXT;
+                keyType = COLUMN_KEY;
+                break;
+            case OPERATOR:
+                displayTextType = OPERATOR_DISPLAY_TEXT;
+                keyType = OPERATOR_KEY;
+                break;
+            case VALUE:
+                displayTextType = VALUE_DISPLAY_TEXT;
+                keyType = VALUE_KEY;
+                break;
+            default:
+                break;
         }
 
-        update(conditionIndex, condition, displayText, displayTextType, fieldId, value, keyType, valueType);
+        update(index, condition, displayText, displayTextType, fieldId, value, keyType, type);
     };
 
     const getFormattedOptions = (options: Array<Object>): any[] => {
         return options.map(option => {
+            const { displayName, id, type } = option;
             return {
-                displayText: option.displayName,
-                fieldId: option.id,
-                type: option.type,
-                value: option.displayName,
+                displayText: displayName,
+                fieldId: id,
+                type,
+                value: displayName,
```

Comment:
I'm trying to decide if value=displayName is expected. But `options` is typed as Objects. 

Is this expected?

Reply:
Is this expected?

Reply:
Yes `value = displayName` is expected. options is typed as Objects because it's possible for it to be either ColumnType, `{displayName: string, id: string, type: string, value: string}`, or `{displayName: string, value: ConnectorType}`.